binderhub/main.py

```diff
@@ -61,6 +61,12 @@ def get(self, provider_prefix, _unescaped_spec):
             org, repo_name, ref = spec.split('/', 2)
             # NOTE: tornado unquotes query arguments too -> notebooks%2Findex.ipynb becomes notebooks/index.ipynb
             filepath = self.get_argument('filepath', '').lstrip('/')
+           
+            # Check if we have a JupyterLab + file path, if so then use it for the filepath
+            urlpath = self.get_argument('urlpath', '').lstrip('/')
+            if 'lab/tree/' in urlpath:
```

Comment:
```suggestion
            if urlpath.startswith("lab/tree"):
```
would this be an improvement/less prone to matching accidentally?

Reply:
I was originally doing it this way to allow for [workspaces in the url too](https://jupyterlab.readthedocs.io/en/stable/user/urls.html#combining-url-functions), but upon looking it seems like that'd actually be a more complicated rule. What if instead we did something like:

```
if urlpath.startswith("lab") and `/tree/` in urlpath:
```

and then do the split on `/tree/` and take whatever is to the right of the first instance of `/tree/`?

Reply:
this is in the latest push, it seemed to work for me locally (both regular lab URLs and workspace URLs)

Reply:
What different kinds of URLs could we get here? I am not really familiar with what Jupyter Lab does/what workspaces are :-/

Reply:
See the link I sent above, that's what I was using to figure out what urls to expect